core/src/main/scala/kafka/log/UnifiedLog.scala

```diff
@@ -687,8 +687,10 @@ class UnifiedLog(@volatile var logStartOffset: Long,
           producerStateManager.updateParentDir(dir)
           // re-initialize leader epoch cache so that LeaderEpochCheckpointFile.checkpoint can correctly reference
           // the checkpoint file in renamed log directory
-          initializeLeaderEpochCache()
-          initializePartitionMetadata()
+          if (reinitialize) {
+            initializeLeaderEpochCache()
```

Comment:
If we don't need to initialize, it's probably safer to set leaderEpochCache to None. This prevents a concurrent append from flushing the leader epoch cache on an invalid file, which could cause the broker to shut down.

Reply:
That's a good point. I've set the leaderEpochCache to None, and also the partitionMetadataFile field after converting it to an Option.